workflow/validate/validate.go

```diff
@@ -865,8 +865,10 @@ func validateArgumentsValues(prefix string, arguments wfv1.Arguments, allowEmpty
 			if len(param.Enum) == 0 {
 				return errors.Errorf(errors.CodeBadRequest, "%s%s.enum should contain at least one value", prefix, param.Name)
 			}
-			if param.Value == nil {
+			if param.Value == nil && !allowEmptyValues {
```

Comment:
I think this would read easier as:
```go
if param.Value == nil {
  if allowEmptyValues {
    return nil
  }
  return errors.Errorf(errors.CodeBadRequest, "%s%s.value is required", prefix, param.Name)
}
```

Reply:
Sounds good, I'll make that change.